htmx-pwa-frontend/lib/render-element.ts

```diff
@@ -0,0 +1,19 @@
+import html from 'html-template-tag';
+import type { Element } from '../types';
+
+export const renderElement = (el: Element): string => {
+  if (typeof el === 'string') return el;
+
+  const attrString = el.shape?.attrs?.join(' ') || '';
+  const { triggers, resource } = el.behavior || {};
+  const triggerString = !triggers
+    ? ''
+    : Array.isArray(triggers)
+      ? `hx-trigger=${triggers.join(', ')} `
+      : `hx-trigger=${triggers} `;
+  const resourceString = !resource ? '' : `hx-${resource.action}=${resource.url}`;
+  const children = el.shape?.children || [];
+  const childrenString =
+    Array.isArray(children) && children.length > 0 ? children.map(renderElement).join('') : '';
```

Comment:
ew, is this htmx specific?

Reply:
is this a one time change or something required per pr?

Reply:
As I discover more htmx behaviors I'll probably have to keep extending the Element type. In theory extending the type should be just extending the function, which shouldn't hurt anyone

Reply:
i see ok per type that is tolerable thaks for answering